lib/blockscore/collection.rb

```diff
@@ -133,8 +133,7 @@ def create(params = {})
     # @api public
     def retrieve(id)
       each do |item|
-        next unless item.id == id
-        return item
+        return item if item.id.eql?(id)
```

Comment:
I realize you're only modifying one part of this class, but I question the design of this collection class. Also, I know you didn't design this so I'm speaking to whomever designed it:

I don't think it should inherit from an Array; you now have an entire super-wide interface that your object needs to adhere to. If you ever want to refactor the object you have to re-implement the dozens of methods in Array that it implements. A much better idea for a collection class is to have a has-a relationship to an Array or Hash under the hood. You can still make it Enumerable as long as you design an `#each` method (I can help with this), and you get to control *exactly* what other methods you want to expose in your public interface.

Also, given that this method does a lookup based on an `id`, and it does so by looping over the entire collection points to the wrong data structure being used. What I would do is change this to have-a `Hash` underneath, where the key is the id, and the value is the object. You can lookup the item quickly using `Hash#fetch`, and still have `#each`:

```ruby
include Enumerable

# ...

def each(&block)
  return to_enum unless block
  items.each_value(&block)
  self
end
```